src/FSAL/FSAL_CORTXFS/mds.c

```diff
@@ -269,25 +280,147 @@ int kvsfs_pmds_ini(struct kvsfs_fsal_module *kvsfs,
+	if (strncmp(pNFS_role,"MDS",sizeof("MDS")) == 0) {
+		mds_ctx->pnfs_role = CORTXFS_PNFS_MDS;
+	} 
+	else if (strncmp(pNFS_role,"DS",sizeof("DS")) == 0) { 
+		mds_ctx->pnfs_role = CORTXFS_PNFS_DS;
+	}
+	else if (strncmp(pNFS_role,"BOTH",sizeof("BOTH")) == 0) {
+		mds_ctx->pnfs_role = CORTXFS_PNFS_BOTH;
+	}
+	else {
+		mds_ctx->pnfs_role = CORTXFS_PNFS_DISABLED;
+	}
+	
+	
+	if (mds_ctx->pnfs_role > CORTXFS_PNFS_DS )
+	{	
+		mds_ctx->mds = gsh_calloc(1, sizeof(struct cortxfs_pnfs_mds_info));
+		if (mds_ctx->mds == NULL) {
+			LogCrit(COMPONENT_PNFS,
 			"gsh_calloc failed: %u", sizeof(*mds_ctx->mds));
-		rc = -ENOMEM;
-		goto out;
+			rc = -ENOMEM;
+			goto out;
+		}
+		mds_ctx->mds->stripe_unit = CORTXFS_PNFS_STRIPE_UNIT;	
+	
+		rc = get_config_item("pNFS","NumDataServers",cfg_items, &items);
+		if (items == NULL){
+			LogCrit(COMPONENT_PNFS,"Zero data servers\n");
+			rc = -rc; 
+			goto out;
+		}
+		
+		num_ds=(int)get_int_config_value(items, 0, 0, NULL);
+		
+		LIST_INIT(&mds_ctx->mds->mds_ds_list);
+			
+		char DS_List[256],tmp[256];
+                char *DS_Info,*port,*ipaddr;
+		int i;	
+	
+		for (i=1; i<=num_ds; i++)
+		{
+			snprintf(DS_List,sizeof(DS_List),"DS%d",i);
+			LogDebug(COMPONENT_PNFS,"DS_List is %s",DS_List);
+			rc = get_config_item("pNFS",DS_List,cfg_items, &items);
+			if (items == NULL)
+			{
+				LogCrit(COMPONENT_PNFS,"No Data Server Entry DS%d",i);
+				rc = -rc; 
+				goto out;
+			}
+			
+			ds = gsh_calloc(1, sizeof(*ds));
+			if (ds == NULL) {
+				LogCrit(COMPONENT_PNFS, "gsh_calloc failed: %u", sizeof(*ds));
+				rc = -ENOMEM;
+				goto out;
+			}			
+ 			/**
+                         * TODO: Once dynamic add/remove DS support is added
+                         * either via temp. config file or via HA, we will need
+                         * to design how to update and maintain ds_id
+                         */
+                        ds->ds_id = i;
+                        LIST_INIT(&ds->ds_addr_list);
+
+                        // per DS, only one addr as of now
+                        // TODO: support for multiple addresses
+                        ds_addr = gsh_calloc(1, sizeof(*ds_addr));
+                        if (ds_addr == NULL) {
+                                LogCrit(COMPONENT_PNFS,
+                                        "gsh_calloc failed: %u",
+                                        sizeof(*ds_addr));
+                                rc = -ENOMEM;
+                                goto out;
+                        }
+                        DS_Info = get_string_config_value(items,NULL);
+			if (DS_Info == NULL){
+				LogCrit(COMPONENT_PNFS,
+					"No valid IP address or Port found\n");
+				goto out;
+			}
+			/* Check if the destination tmp buffer is big enough */
+			if (sizeof(DS_Info) > sizeof(tmp)) {
+				LogCrit(COMPONENT_PNFS,
+					"Destination buffer too small\n");
+				goto out;
+			}
+			/* Get the IP address */
+			strcpy(tmp,DS_Info);
```

Comment:
pls use snprintf or strncpy

Reply:
Codacy has strncpy isn its list of banned functions.I have used strcpy with a check for length